src/stratis/buff_log.rs

```diff
@@ -0,0 +1,195 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//! A logger that wraps another logger, and provides "flight data recorder"
+//! semantics: nothing is output until asked for. This can be by calling
+//! `Handle::dump()`, or a `HandleGuard` can be used to dump the log when a
+//! scope is exited.
+
+use std::collections::VecDeque;
```

Comment:
I couldn't get a really good handle on how this works, so I just have two questions about the whole thing:
1. Why do you need a Mutex at all, much less two, when stratisd is single threaded?
2. If you do need a Mutex, you shouldn't be using an ```expect()``` on the result of ```lock()```. You can't prove that it will succeed unless you can globally prove no panics in all of stratisd which is a stretch. The docs suggest using an ```unwrap()``` to panic every thread that tries to access the shared data underneath the mutex. That's a simple approach, and if you documented that that's the one you want to take, then a bunch of ```unwrap()``` make sense. But since this is a log, and not something that is totally central to stratisd's correct operation, it is almost certainly better to do something different than just panic the thread, AFAICT.

Reply:
1. All impls of trait `Log` must also be `Send` and `Sync`. so the log macros can be used from anywhere. I think just making it always threadsafe was judged easier than having a threadsafe and a non-threadsafe version of the trait. The second one is because we're also sharing our log implementation at a higher level (via the `Handle`) and because Handle is parameterized over `Log`, it also needs to be `Send` and `Sync`. Not 100% on this reasoning but it makes sense, and the compiler won't let me remove anything so I guess it knows best :smiley: 

1. The unwrap (or expect) will only fail if the other thread paniced while holding the mutex. The only code that holds these mutexes are in this file.